coffee/chosen.jquery.coffee

```diff
@@ -76,8 +77,10 @@ class Chosen extends AbstractChosen
 
     @container.on 'mousedown.chosen', (evt) => this.container_mousedown(evt); return
     @container.on 'mouseup.chosen', (evt) => this.container_mouseup(evt); return
-    @container.on 'mouseenter.chosen', (evt) => this.mouse_enter(evt); return
-    @container.on 'mouseleave.chosen', (evt) => this.mouse_leave(evt); return
+    @container.on 'focusin.chosen', (evt) => this.container_focusin(evt); return
+    @container.on 'focusout.chosen', (evt) => this.container_focusout(evt); return
+    @container.on 'chosen:blur.chosen', (evt) => this.close_field(evt); return
+    @container.on 'chosen:focus.chosen', (evt) => this.input_focus(evt); return
```

Comment:
I think relying on custom events here (`chosen:blur` and `chosen:focus`) are the most straightforward way to handle this — but we can also just simply inline the calls to `input_focus` and `close_field` in the `container_focusin` and `container_focusout` events below. Let me know which you prefer!

Reply:
This suggests that those events can/should be invoked from a client implementation, which I think is not the case, right? So I think inlining them below would be better.